### ISSUE TITLE
bug: Ibis returns quoted strings when selecting and casting JSON values to strings from a DuckDB table

### ISSUE THREAD

**Issue by dixdotdev** [ibis-project/ibis#8948]
### What happened?

When selecting JSON data from a DuckDB table and casting it to a string it returns the data in quotes.

```python
import ibis
ibis.options.interactive = True

ddb = ibis.duckdb.connect("mydb.duckdb")

schema = ibis.schema( {
    "id": "int64",
    "name": "string",
    "json_data": "json",
})

ddb.create_table("mytable", schema=schema)

data = [
    {"id": 1, "name": "Alice", "json_data": {"last_name": "Smith", "age": 40}},
    {"id": 2, "name": "Bob", "json_data": {"last_name": "Jones", "age": 39}},
    {"id": 3, "name": "Charlie", "json_data": {"last_name": "Davies", "age": 54}},
]

ddb.insert("mytable", data)

t = ddb.table("mytable")

t

┏━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┓
┃ id    ┃ name    ┃ json_data                          ┃
┡━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━┩
│ int64 │ string  │ json                               │
├───────┼─────────┼────────────────────────────────────┤
│     1 │ Alice   │ {'last_name': 'Smith', 'age': 40}  │
│     2 │ Bob     │ {'last_name': 'Jones', 'age': 39}  │
│     3 │ Charlie │ {'last_name': 'Davies', 'age': 54} │
└───────┴─────────┴────────────────────────────────────┘

t.select(
    t.json_data["last_name"].cast("string").name("last_name"),
    t.json_data["age"].cast("int64").name("age"),
)

┏━━━━━━━━━━━┳━━━━━━━┓
┃ last_name ┃ age   ┃
┡━━━━━━━━━━━╇━━━━━━━┩
│ string    │ int64 │
├───────────┼───────┤
│ "Smith"   │    40 │
│ "Jones"   │    39 │
│ "Davies"  │    54 │
└───────────┴───────┘
```

I would expect this to return:

```python
┏━━━━━━━━━━━┳━━━━━━━┓
┃ last_name ┃ age   ┃
┡━━━━━━━━━━━╇━━━━━━━┩
│ string    │ int64 │
├───────────┼───────┤
│ Smith     │    40 │
│ Jones     │    39 │
│ Davies    │    54 │
└───────────┴───────┘
```

Work around is to perform the following command instead:
```python
t.select(
    t.json_data["last_name"].cast("string").replace('"', "").name("last_name"),
    t.json_data["age"].cast("int64").name("age"),
)
```

### What version of ibis are you using?

8.0.0

### What backend(s) are you using, if any?

DuckDB

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by gforsyth** [comment]
Thanks for reporting this @jimdix!

We will think about how to proceed with this one.  Our first step will be to sort out whether or not we are being consistent in how we handle JSON across the various backends that support it.
Next up, we have to think about what the "right thing" (or at least, the "consistent thing") to do is when casting to 'string' from a valid JSON string to something that a human would want to see.

**Comment by jimdix** [comment]
Thanks for the response @gforsyth. 

With regards to the right thing, I would argue that adding ``""`` to the values from the JSON isn't accurately representing the value from the JSON. If it's casted to a string type column, then ``value`` is materially different to ``"value"``, especially when you want to ``JOIN`` or filter on it after the fact.

**Comment by cpcloud** [comment]
Here's the current list of backends that nominally support a JSON type and thus could in theory support JSON operations:

- BigQuery
- DuckDB
- MySQL
- PostgreSQL
- RisingWave
- Snowflake
- SQLite
- Trino

Some of these implement our (currently quite small) JSON API (`__getitem__`, `map`, and `array`) and some don't.

**Comment by cpcloud** [comment]
Closed by #8958.
